cni-plugin/main.go

```diff
@@ -249,9 +250,18 @@ func cmdAdd(args *skel.CmdArgs) error {
 			}
 
 			if pod.GetLabels()[k8s.ControllerComponentLabel] != "" {
-				// Skip 443 outbound port if its a control plane component
-				logEntry.Debug("linkerd-cni: adding 443 to OutboundPortsToIgnore as its a control plane component")
-				options.OutboundPortsToIgnore = append(options.OutboundPortsToIgnore, "443")
+				// Skip k8s api server ports on the outbound side if pod is a
+				// control plane component
+				skippedPorts, err := getApiServerPorts(ctx, client)
+				if err != nil {
+					// If we cannot retrieve the 'kubernetes' service's ports (for
+					// whatever reason), skip default ports: 443, 6443
+					logEntry.Debugf("linkerd-cni: could not retrieve ports from 'kubernetes' service: %v", err)
```

Comment:
probably worth logging this at Error level instead of Debug.

Reply:
Nice, done!